src/secex-genprotimgvm-scripts/genprotimg-script.sh

```diff
@@ -1,30 +1,63 @@
 #!bin/bash
-
 set -exuo pipefail
 
 echo "Preparing for genprotimg-daemon"
 
-source="/build/genprotimg"
-destination="/genprotimg"
-pkey="usr/lib/coreos/ignition.asc"
+PREFIX=/
+if [ -e /var/genprotimg ]; then
+    PREFIX=/var
+fi
+
+basedir=$(pwd)
+destination="${PREFIX}/genprotimg"
+parmfile="${PREFIX}/build/parmfile"
+
+# This is our coreos.qemu-secex.qcow2
+disk=$(realpath /dev/disk/by-id/virtio-target)
+
+# We pass Ignition gpg private key from COSA to the VM as virtual disk
+gpg_private_key=$(realpath /dev/disk/by-id/virtio-gpgkey)
+gpg_dir=$(mktemp -d)
 
-trap "rm -f ${source}/${pkey}" EXIT
+# 'boot' labeled partition on target image
+disk_boot="${disk}3"
+boot_mnt=$(mktemp -d /tmp/boot-XXXXXX)
+
+mount -o ro "${disk_boot}" "${boot_mnt}"
+
+cleanup() {
+    cd "${basedir}"
+    rm -rf ${gpg_dir}/*
+    umount -q ${boot_mnt}
+}
+trap "cleanup" EXIT
```

Comment:
I feel like we can get rid of this and end up with easier to reason code. The key is available as a block device so there's no use trying to get rid of it quickly within this VM.

E.g. WDYT about reflowing as:

```bash
boot_mnt=$(mktemp -d /tmp/boot-XXXXXX)
mount -o ro "${disk_boot}" "${boot_mnt}"
blsfile=$(ls "${boot_mnt}"/loader/entries/*.conf)
kernel=$(grep linux "${blsfile}" | cut -d' ' -f2)
initrd=$(grep initrd "${blsfile}" | cut -d' ' -f2)
cp "$PWD/${kernel}" "${destination}/vmlinuz"
cp "$PWD/${initrd}" "${destination}/initrd.img")
umount "${boot_mnt}"
rmdir "${boot_mnt}"
```

And

```bash
gpg_dir=$(mktemp -d)
(cd "${gpg_dir}"
 mkdir -p "usr/lib/coreos"
 pkey="usr/lib/coreos/ignition.asc"
 cat "${gpg_private_key}" > "${pkey}"
 echo "${pkey}" | cpio --quiet -H newc -o | gzip -9 -n >> "${destination}/initrd.img")
rm -rf "${gpg_key}"
```

That moves the declaring of variables, creation of tmpdirs, and associated cleanups closer together. I also like avoiding `cd` if possible, or otherwise spawn subshells for it so it's obvious that it doesn't affect the rest of the code.